dev/src/expression.ts

```diff
@@ -0,0 +1,6739 @@
+   */
+  static of(value: undefined): Constant;
+
+  /**
+   * Creates a `Constant` instance for a GeoPoint value.
+   *
+   * @param value The GeoPoint value.
+   * @return A new `Constant` instance.
+   */
+  static of(value: firestore.GeoPoint): Constant;
+
+  /**
+   * Creates a `Constant` instance for a Timestamp value.
+   *
+   * @param value The Timestamp value.
+   * @return A new `Constant` instance.
+   */
+  static of(value: firestore.Timestamp): Constant;
+
+  /**
+   * Creates a `Constant` instance for a Date value.
+   *
+   * @param value The Date value.
+   * @return A new `Constant` instance.
+   */
+  static of(value: Date): Constant;
+
+  /**
+   * Creates a `Constant` instance for a Uint8Array value.
+   *
+   * @param value The Uint8Array value.
+   * @return A new `Constant` instance.
+   */
+  static of(value: Uint8Array): Constant;
+
+  /**
+   * Creates a `Constant` instance for a DocumentReference value.
+   *
+   * @param value The DocumentReference value.
+   * @return A new `Constant` instance.
+   */
+  static of(value: firestore.DocumentReference): Constant;
+
+  /**
+   * Creates a `Constant` instance for a Firestore proto value.
+   *
+   * @param value The Firestore proto value.
+   * @return A new `Constant` instance.
+   */
+  static of(value: api.IValue): Constant;
+
+  /**
+   * Creates a `Constant` instance for an array value.
+   *
+   * @param value The array value.
+   * @return A new `Constant` instance.
+   */
+  static of(value: Array<any>): Constant;
+
+  /**
+   * Creates a `Constant` instance for a map value.
+   *
+   * @param value The map value.
+   * @return A new `Constant` instance.
+   */
+  static of(value: Map<string, any>): Constant;
+
+  /**
+   * Creates a `Constant` instance for a VectorValue value.
+   *
+   * @param value The VectorValue value.
+   * @return A new `Constant` instance.
+   */
+  static of(value: firestore.VectorValue): Constant;
+
+  /**
+   * Creates a `Constant` instance for a Firestore proto value.
+   *
+   * @param value The Firestore proto value.
+   * @return A new `Constant` instance.
+   */
+  static of(value: api.IValue): Constant;
+  static of(value: any): Constant {
+    return new Constant(value);
+  }
+
+  /**
+   * Creates a `Constant` instance for a VectorValue value.
+   *
+   * ```typescript
+   * // Create a Constant instance for a vector value
+   * const vectorConstant = Constant.ofVector([1, 2, 3]);
+   * ```
+   *
+   * @param value The VectorValue value.
+   * @return A new `Constant` instance.
+   */
+  static vector(value: Array<number> | firestore.VectorValue): Constant {
+    if (value instanceof VectorValue) {
+      return new Constant(value);
+    } else {
+      return new Constant(new VectorValue(value as Array<number>));
+    }
+  }
+
+  _toProto(serializer: Serializer): api.IValue {
+    if (isFirestoreValue(this.value)) {
+      return this.value;
+    }
+
+    return serializer.encodeValue(this.value)!;
```

Comment:
`serializer.endcodeValue(undefined)` will return `null`, which is an permitted, but unhandled case. Do we want to treat undefined as null?